packages/visual-editor/src/components/puck/Header.tsx

```diff
@@ -85,97 +50,139 @@ const headerFields: Fields<HeaderProps> = {
 
   return (
     <header
-      className={themeManagerCn(
-        "w-full bg-white components",
-        headerVariants({ backgroundColor })
-      )}
+      className={`components font-body-fontFamily relative ${headerVariants({ backgroundColor })}`}
     >
-      <div className="mx-auto flex max-w-6xl flex-1 items-center justify-between px-4 py-6">
-        {resolvedLogo && (
+      <div className="container mx-auto py-5 flex justify-start md:justify-between px-4 sm:px-8 lg:px-16 xl:px-20 items-center">
+        {logo && (
           <EntityField
-            displayName="Logo"
-            fieldId={logo.image.field}
-            constantValueEnabled={logo.image.constantValueEnabled}
+            displayName="Business Logo"
+            fieldId={"site.businessLogo"}
           >
-            <Image
-              image={resolvedLogo}
-              className="w-full h-full object-cover"
-            />
+            <HeaderLogo logo={logo} logoLink={logoLink} />
           </EntityField>
         )}
-        <div className="flex items-center justify-end space-x-4">
-          <ul className="flex space-x-8">
-            {resolvedLinks?.map((item, idx) => (
-              <li
-                key={idx}
-                className="cursor-pointer font-bold text-header-linkColor text-header-linkFontSize hover:underline"
-              >
-                {item.link && (
-                  <EntityField
-                    displayName="Link"
-                    fieldId={links[idx].cta.field}
-                    constantValueEnabled={links[idx].cta.constantValueEnabled}
-                  >
-                    <Link cta={item} eventName={`header-link-${item.label}`} />
-                  </EntityField>
-                )}
-              </li>
-            ))}
-          </ul>
-        </div>
+
+        <EntityField displayName="Header Links" fieldId={"site.header.links"}>
+          <HeaderLinks links={links} />
+        </EntityField>
+
+        <button
+          className="flex md:hidden absolute p-4 right-0 top-1/2 -translate-y-1/2"
+          onClick={() => setMenuOpen(!menuOpen)}
+          aria-label={menuOpen ? "Close header menu" : "Open header menu"}
+        >
+          {menuOpen ? <FaTimes size={"1.5rem"} /> : <FaBars size={"1.5rem"} />}
+        </button>
       </div>
+
+      <HeaderMobileMenu
+        isOpen={menuOpen}
+        links={links}
+        backgroundColor={backgroundColor}
+      />
     </header>
   );
 };
+
+const HeaderLogo = (props: { logo: ComplexImageType; logoLink?: string }) => {
+  return (
+    <MaybeLink href={props.logoLink}>
+      <div className="flex mr-2">
+        <Image
+          image={props.logo}
+          layout="fixed"
+          height={props.logo.image.height}
+          width={props.logo.image.width}
+        />
+      </div>
+    </MaybeLink>
+  );
+};
+
+const HeaderLinks = (props: { links: CTA[] }) => {
+  return (
+    <div className="hidden md:flex items-center">
```

Comment:
I had a similar comment on Alexis' PR. If there are no items we shouldn't render the div at all. You'll need to check that everything else is styled correctly in this case. And same for any other fields, like no logo. I also imagine if there are no links at all then we wouldn't want to render the mobile menu thing.